conjur/api/client.py

```diff
@@ -125,7 +127,10 @@ def __init__(self,
             self._api = Api(http_debug=http_debug,
                             ssl_verify=ssl_verify,
                             **loaded_config)
-            self._api.login(login_id, password)
+            if service_id is not None:
```

Comment:
A `service_id` seems to be a common attribute in all our authenticators not just LDAP.  Because of this, IMO we should distinguish LDAP specifically from other authenticators instead of just through identifying a service_id so we don't have to perform a refactor in the future. Maybe add another function `distinguish_authentication_method` where there we evaluate the value of the supported authentication method (`authn`, `authn-ldap`, etc)